src/WebJobs.Script/Description/DotNet/FunctionAssemblyLoadContext.cs

```diff
@@ -4,63 +4,147 @@
+                return LoadFromAssemblyPath(path);
             }
+
+            return null;
+        }
+
+        internal bool TryLoadAssembly(AssemblyName assemblyName, out Assembly assembly)
+        {
+            assembly = Load(assemblyName);
+
+            return assembly != null;
         }
 
-        public Assembly FunctionAssembly { get; private set; }
+        protected override IntPtr LoadUnmanagedDll(string unmanagedDllName)
+        {
+            // For now, we'll attempt to resolve unmanaged DLLs from the base probing path
+
+            // Directory resolution is be simple for now, we'll assume the base probing
+            // path (usually, the bin folder in the function app root) and combine with
+            // the file name resolved from the native DLL reference:
+            string fileName = GetUnmanagedLibraryFileName(unmanagedDllName);
 
-        public FunctionMetadata Metadata { get; private set; }
+            if (File.Exists(fileName))
+            {
+                LoadUnmanagedDllFromPath(fileName);
+            }
 
-        internal Assembly ResolveAssembly(string name)
+            return base.LoadUnmanagedDll(unmanagedDllName);
+        }
+
+        internal string GetUnmanagedLibraryFileName(string unmanagedLibraryName)
         {
-            Assembly assembly = _metadataResolver.ResolveAssembly(name);
+            // We need to properly resolve the native library in different platforms:
+            // - Windows will append the '.DLL' extension to the name
+            // - Linux uses the 'lib' prefix and '.so' suffix. The version may also be appended to the suffix
+            // - macOS uses the 'lib' prefix '.dylib'
+            // To handle the different scenarios described above, we'll just have a pattern that gives us the ability
+            // to match the variations across different platforms. If needed, we can expand this in the future to have
+            // logic specific to the platform we're running under.
+
+            string fileName = null;
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                if (unmanagedLibraryName.EndsWith(".dll", StringComparison.OrdinalIgnoreCase))
+                {
+                    fileName = unmanagedLibraryName;
+                }
+                else
+                {
+                    fileName = unmanagedLibraryName + ".dll";
+                }
+            }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            {
+                fileName = "lib" + unmanagedLibraryName + ".dylib";
+            }
+            else
+            {
+                fileName = "lib" + unmanagedLibraryName + ".so";
+            }
 
-            if (assembly != null)
+            return fileName;
+        }
+
+        protected static string ResolveFunctionBaseProbingPath()
+        {
+            string basePath = null;
+            var settingsManager = ScriptSettingsManager.Instance;
```

Comment:
Yes! The goal is for all of that to be gone soon. I'll actually make a quick change to avoid that altogether now since it does make sense for this just to read environment variables.